### DISCUSSION TITLE
关于zkp在full node 同步中的可以做的优化的原理？

### DISCUSSION THREAD

**Discussion by cuiweixie** [z2o-k7e/halo2-step-by-step#8] [Q&A] [answered]
full node 同步过程中，我理解，可以使用zkp证明执行到最新区块的正确性，而不需要下载执行重创世块到当前最新快的历史数据。
我的理解是这样的，以full node P, 新加入的full node V举例：
1. P 生成到最新块的证明，发给V。
2. V确认证明是正确的。
3. V向P申请同步当前的世界状态。
4. P发送世界状态给V。 同步完成。
疑问点在于，第四步， P向v发送数据的时候，可以作假，伪造数据。
更一般的说，P向V证明了某个事实。然后给V发送数据，怎么确保V不造假。

**Comment by qizhou** [comment]
3的b是什么？

**Reply by cuiweixie** [reply]
应该是P. 已修改？

**Reply by qizhou** [reply]
如果是基于ZKP的stateless client，V不需要世界状态，V只需要世界状态的digest就可以了。

**Reply by cuiweixie** [reply]
digest?是新的full node 启动的时候，是否能做到只下载当前最新的世界状态，不同步历史区块。不算是stateless client？

**Reply by qizhou** [reply, marked as answer]
digest就是世界状态的hash (state trie root).

**Reply by voidcenter** [reply]
如果只要最新块的状态，应该也可以把 latest block 和 digest 共同作为 public output 就可以了是吧？
